lib/checkPullRequestReview.js

```diff
@@ -181,52 +213,31 @@ const handleApproval = async (context, pullRequest, review) => {
     );
   }
 
-  // Check if pull request has been approved by all reviewers.
-  // The pull request data does not contain information if the pull request
-  // has been approved, so we are searching for an approved pull request
-  //  with the pull request number and in the appropriate repository. An
-  // empty array will be returned if the pull request is not approved by
-  // all reviewers.
-  const repoSearchString = `repo:${context.payload.repository.full_name}`;
-  const reviewApprovedSearchString = 'review:approved';
-  const searchResult = await context.github.search.issuesAndPullRequests(
-    context.repo({
-      q:
-        repoSearchString +
-        ' ' +
-        reviewApprovedSearchString +
-        ' ' +
-        pullRequest.number,
-    })
+  // Check if the PR has pending reviews and assign remaining reviewers.
+  // If the PR does not have any pending review, check if it has changes
+  // requested and do nothing since the PR author was already assigned when
+  // the review was made.
+  // If none of the above cases, check if the PR has been approved (the search mechanism)
```

Comment:
Explain the search mechanism. Also, add a full stop in end.

Reply:
Done.